src/Security/UserChecker.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace AppBundle\Security;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Exception\AccountNotValidatedException;
+use Symfony\Component\Security\Core\Exception\DisabledException;
+use Symfony\Component\Security\Core\User\UserCheckerInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class UserChecker implements UserCheckerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function checkPreAuth(UserInterface $user)
+    {
```

Comment:
What about check user validation status here and skip the auth check if is not valid user?

Reply:
check with you ;)

Reply:
yes, thanks 👍

Reply:
Curious, why not ?

Reply:
For the record, it's because we only give hints to users that gives the right credentials